src/components/CustomDropdown/index.tsx

```diff
@@ -22,16 +29,110 @@ const CustomDropdown: React.FC<CustomDropdownProps> = ({
     placement,
     disabled,
 }) => {
+    const [dropdownState, setDropdownState] = useState<DropdownState>(
+        DropdownState.CLOSED
+    );
+
+    const triggerRef = useRef<HTMLElement | null>(null);
+    const dropdownRef = useRef<HTMLDivElement | null>(null);
+    const closeTimeoutRef = useRef<NodeJS.Timeout | null>(null);
+
+    /**
+     * Prevents the menu wrapper from capturing focus,
+     * this prevents losing focus to the body
+     * when "Escape" is pressed.
+     */
+    useEffect(() => {
+        const element = dropdownRef.current;
+        if (element) {
+            const menuElement = element.querySelector(
+                ".ant-dropdown-menu"
+            ) as HTMLElement;
+            if (menuElement) {
+                if (dropdownState === DropdownState.FORCED_OPEN) {
+                    menuElement.setAttribute("tabIndex", "-1");
+                } else if (dropdownState === DropdownState.CLOSED) {
+                    menuElement.setAttribute("tabIndex", "0");
+                }
+            }
+        }
+    }, [dropdownRef, dropdownState]);
+
+    /**
+     * Manually handling keydown and hover behavior because
+     * our focus management overrides the defaults of the antd components.
+     */
+    const openTriggers = new Set(["Enter", " ", "ArrowDown"]);
+
+    const handleKeyDown: KeyboardEventHandler<any> = (event) => {
+        if (event.key === "Escape") {
+            event.preventDefault();
+            setDropdownState(DropdownState.CLOSED);
+            triggerRef.current?.focus();
+        }
+        if (
+            openTriggers.has(event.key) &&
+            dropdownState === DropdownState.CLOSED
+        ) {
+            event.preventDefault();
+            setDropdownState(DropdownState.FORCED_OPEN); // Opened by keyboard
```

Comment:
May need to also clear `closeTimeoutRef` here. I think it's possible to mouse over the menu quickly while it's closed, which would start a close timer, and then use the keyboard to click on the menu to get it into the FORCED_OPEN state. Sincere there's no check for FORCED_OPEN when the timeout fires, it would suddenly close the menu.

Reply:
So this is true, and also made me realize that focus was being captured in that scenario because of `autoFocus`.

This implies some things to deal with in the next PR, but for now removing `autoFocus`, clearing the timeout here, also running this anytime the state is not `FORCED_OPEN` is doing the trick.